tikv-client/src/main/java/com/pingcap/tikv/types/DateType.java

```diff
@@ -47,6 +49,28 @@ public Date getOriginDefaultValueNonNull(String value, long version) {
     return Converter.convertToDate(value);
   }
 
+  @Override
+  protected Object doConvertToTiDBType(Object value)
+      throws ConvertNotSupportException, ConvertOverflowException {
+    return convertToMysqlDate(value);
+  }
+
+  private java.sql.Date convertToMysqlDate(Object value) throws ConvertNotSupportException {
+    java.sql.Date result;
+    if (value instanceof Long) {
+      result = new java.sql.Date((Long) value);
+    } else if (value instanceof String) {
+      result = java.sql.Date.valueOf((String) value);
+    } else if (value instanceof java.sql.Date) {
+      result = (java.sql.Date) value;
+    } else if (value instanceof java.sql.Timestamp) {
+      result = new java.sql.Date(((java.sql.Timestamp) value).getTime());
```

Comment:
We may consider the effect of timezone.

Reply:
would you please provide a test case, which is wrong for the current code?

Reply:
do we have unit tests for `convertToMysqlDate()`?

Reply:
disable all timezone related conversion

Reply:
@birdstorm  see `ToDateSuite`

Reply:
Gotcha. Let me fix timezone related issue = =

Reply:
awesome